library/tasks/HISAT2.wdl

```diff
@@ -0,0 +1,301 @@
+      --seed 12345 \
+      -k 10 \
+      --secondary \
+      -p ${cpu} -S ${output_name}.sam
+    samtools sort -@ ${cpu} -O bam -o "${output_name}.bam" "${output_name}.sam"
+  }
+
+  runtime {
+    docker: docker
+    memory: machine_mem_mb + " MB"
+    disks: "local-disk " + disk + " HDD"
+    cpu: cpu
+    preemptible: preemptible
+  }
+
+  output {
+    File logfile = "${output_name}.log"
+    File metfile = "${output_name}.hisat2.met.txt"
+    File output_bam = "${output_name}.bam"
+  }
+}
+
+task HISAT2RSEM {
+  meta {
+    description: "JISHUUUUUUUUUUUUUUUU"
+  }
+
+  parameter_meta {
+    hisat2_ref: ""
+    fq1: "gz forward fastq file"
+    fq2: "gz reverse fastq file"
+    ref_name: ""
+    output_name: ""
+    sample_name: ""
+    docker: ""
+    memory_gb: ""
+    cpu: ""
+    disk: ""
+    preemptible: ""
+  }
+
+  File hisat2_ref
+  File fq1
+  File fq2
+  String ref_name
+  String output_name
+  String sample_name
+
+  # runtime values
+  String docker = "quay.io/humancellatlas/secondary-analysis-hisat2:v0.2.2-2-2.1.0"
+  Int memory_gb = 5
+  Int machine_mem_mb = memory_gb * 1000
+  Int cpu = 4
+  # use provided disk number or dynmically size on our own, 10 is our zipped fastq -> bam conversion with 50GB of additional disk
+  Int disk = ceil((size(fq1, "GB") + size(fq2, "GB") * 10) + size(hisat2_ref, "GB") + 50)
+  Int preemptible = 5
+
+  command {
+    set -e
+
+    # fix names if necessary.
+    if [ "${fq1}" != *.fastq.gz ]; then
+        FQ1=${fq1}.fastq.gz
+        mv ${fq1} ${fq1}.fastq.gz
+    else
+        FQ1=${fq1}
+    fi
+
+    if [ "${fq2}" != *.fastq.gz ]; then
+        FQ2=${fq2}.fastq.gz
+        mv ${fq2} ${fq2}.fastq.gz
+    else
+        FQ2=${fq2}
+    fi
+
+    tar -zxvf "${hisat2_ref}"
+    hisat2 -t \
+      -x ${ref_name}/${ref_name} \
+      -1 $FQ1 \
+      -2 $FQ2 \
+      --rg-id=${sample_name} --rg SM:${sample_name} --rg LB:${sample_name} \
+      --rg PL:ILLUMINA --rg PU:${sample_name} \
+      --new-summary --summary-file ${output_name}.log \
+      --met-file ${output_name}.hisat2.met.txt --met 5 \
+      -k 10 \
+      --mp 1,1 \
+      --np 1 \
+      --score-min L,0,-0.1 \
+      --secondary \
+      --no-mixed \
+      --no-softclip \
+      --no-discordant \
+      --rdg 99999999,99999999 \
+      --rfg 99999999,99999999 \
+      --no-spliced-alignment \
+      --seed 12345 \
+      -p ${cpu} -S ${output_name}.sam
+    samtools view -bS  "${output_name}.sam" > "${output_name}.bam"
+  }
+
+  runtime {
+    docker: docker
+    memory: machine_mem_mb + " MB"
+    disks: "local-disk " + disk + " HDD"
+    cpu: cpu
+    preemptible: preemptible
+  }
+
+  output {
+    File logfile = "${output_name}.log"
+    File metfile = "${output_name}.hisat2.met.txt"
+    File output_bam = "${output_name}.bam"
+  }
+}
+
+task HISAT2SingleEnd {
+  File hisat2_ref
+  File fq
+  String ref_name
+  String output_name
+  String sample_name
+
+  # runtime values
```

Comment:
This version is the same as the previous one except all the values are optional.  This explicitly tells the user that they can optionally set these values but they do have a default.  Downside to this is that I don't really think optionals are meant to be used this way and you have to add a `select_first()` where ever you do plan to use these values.